semtypes/src/main/java/io/ballerina/types/MappingAlternative.java

```diff
@@ -36,22 +36,22 @@ private MappingAlternative(SemType semType, MappingAtomicType[] pos, MappingAtom
         this.neg = neg;
     }
 
-    public MappingAlternative[] mappingAlternativesRw(Context cx, SemType t) {
-        if (t instanceof UniformTypeBitSet) {
-            if ((((UniformTypeBitSet) t).bitset & PredefinedType.MAPPING_RW.bitset) == 0) {
+    public MappingAlternative[] mappingAlternatives(Context cx, SemType t) {
+        if (t instanceof BasicTypeBitSet) {
+            if ((((BasicTypeBitSet) t).bitset & PredefinedType.MAPPING.bitset) == 0) {
```

Comment:
Similar to above shall we consider using pattern matching here as well
```suggestion
        if (t instanceof BasicTypeBitSet b) {
            if ((b.bitset & PredefinedType.MAPPING.bitset) == 0) {
```